x-pack/plugins/secops/public/components/page/network/network_top_n_flow_table/columns.tsx

```diff
@@ -0,0 +1,233 @@
+              and: [],
+              enabled: true,
+              id,
+              name: domain,
+              excluded: false,
+              kqlQuery: '',
+              queryMatch: {
+                field: domainAttr,
+                value: domain,
+              },
+              queryDate: {
+                from: startDate,
+                to: Date.now(),
+              },
+            }}
+            render={(dataProvider, _, snapshot) =>
+              snapshot.isDragging ? (
+                <DragEffects>
+                  <Provider dataProvider={dataProvider} />
+                </DragEffects>
+              ) : (
+                <>
+                  {domain}
+                  {domains.length > 1 && (
+                    <EuiToolTip
+                      content={
+                        <>
+                          {domains.slice(1, 6).map(domainName => (
+                            <span key={`${id}-${domainName}`}>
+                              {defaultToEmptyTag(domainName)}
+                              <br />
+                            </span>
+                          ))}
+                          {domains.slice(1).length > 5 && (
+                            <b>
+                              <br />
+                              more ...
+                            </b>
+                          )}
+                        </>
+                      }
+                    >
+                      <MoreDomains type="eye" />
+                    </EuiToolTip>
+                  )}
+                </>
+              )
+            }
+          />
+        );
+      } else {
+        return getEmptyTagValue();
+      }
+    },
+  },
+  {
+    name: i18n.DIRECTION,
+    truncateText: false,
+    hideForMobile: false,
+    render: ({ node }) =>
+      has('network.direction', node)
+        ? get('network.direction', node).length > 0
+          ? get('network.direction', node).map((direction: NetworkDirectionEcs) => (
+              <AddToKql
+                key={escapeDataProviderId(
+                  `networkTopNFlow-table-${topNFlowType}-${topNFlowDirection}-direction-${direction}`
+                )}
+                content={i18n.FILTER_TO_KQL}
+                expression={`network.direction: ${escapeQueryValue(direction)}`}
+                type={type}
+              >
+                {defaultToEmptyTag(direction)}
+              </AddToKql>
+            ))
+          : getEmptyTagValue()
+        : getEmptyTagValue(),
+  },
+  {
+    name: i18n.BYTES,
+    truncateText: false,
+    hideForMobile: false,
+    render: ({ node }) =>
+      has('network.bytes', node)
+        ? numeral(node.network!.bytes).format('0.000b')
+        : getEmptyTagValue(),
+  },
+  {
+    name: i18n.PACKETS,
+    truncateText: false,
+    hideForMobile: false,
+    render: ({ node }) =>
+      has('network.packets', node)
+        ? numeral(node.network!.packets).format('0,000')
+        : getEmptyTagValue(),
```

Comment:
Not to bum you out, but what this is doing that is a bit difficult to spot ... This is returning `0` in a lot of cases when really it is getting `null` from the `node.network.packets`. The same thing applies to the other `has` clauses.

For example if `node.network.packets = null` which it is in a lot of cases, then `has('network.packets', node)` is returning true as if it does have a value. The value of `null`. Then when you send null into `numeral(null).format('0,000')` it is returning a 0 back to you but really there are no packets in a lot of cases.

If I change this code to:

```ts
    render: ({ node }) => {
      if (node.network && node.network.packets) {
        return numeral(node.network.packets).format('0,000');
      } else {
        return getEmptyTagValue();
      }
    },
```
I actually get back this UI with double dashes which I think is more accurate since it is null:

<img width="1309" alt="screen shot 2019-02-22 at 4 39 57 pm" src="https://user-images.githubusercontent.com/1151048/53277324-8dac2280-36c0-11e9-9a3c-0378c2eb4dc1.png">

I would look at the other `has` ones as well to ensure they're doing what you expect.